File: ex06/numero_perdido.c
#include<stdio.h>

int main() {
    long long n;
    if (scanf("%lld", &n) != 1) return 0;

    long long soma_esperada = (n * (n + 1)) / 2;
    long long soma_atual = 0;
    long long temp;

    for (int i = 0; i < n - 1; i++) {
        if (scanf("%lld", &temp) == 1) {
            soma_atual += temp;
        }
    }

    printf("%lld\n", soma_esperada - soma_atual);

    return 0;
}

File: ex06/numero_perdido.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();

    // Lê n
    let n: i64 = match lines.next() {
        Some(Ok(line)) => line.trim().parse().unwrap_or(0),
        _ => return,
    };

    if n < 2 {
        return;
    }

    let soma_esperada = (n * (n + 1)) / 2;
    let mut soma_atual: i64 = 0;

    for line in lines {
        if let Ok(l) = line {
            for num_str in l.split_whitespace() {
                if let Ok(num) = num_str.parse::<i64>() {
                    soma_atual += num;
                }
            }
        }
    }

    println!("{}", soma_esperada - soma_atual);
}
